### PR TITLE
Add before-and-after context to JS overview page

### LINKED ISSUE #5630: Add before-and-after context to JS interop overview page
Add a table of the previous solutions and their new counterparts to the front page of the new JS interop documentation

`dart:html`, other web libraries > package:web
`package:js` > `dart:js_interop` and `unsafe` 

Plus a little context on the updates. This will make the page a much better landing for wasm/other migration stories. 

fyi @kevmoo and @srujzs -- I'll take care of this asap since I'd like to use it in the flutter wasm update https://github.com/flutter/website/pull/10249

### PR DESCRIPTION
**Pull request by MaryaBelanger**
Fixes #5630 

@srujzs / @kevmoo please check the wording makes sense / it gets the right message across. Thanks!

### CODE REVIEW

File: src/content/interop/js-interop/index.md [hunk 1]
```diff
@@ -10,12 +10,44 @@ JavaScript apps and libraries, as well as browser APIs, using `dart:js_interop`.
...
+developers access to JavaScript and browser bindings in their Dart code.
+This next generation of web interop not only improves user experience,
+but also enables [Wasm][] support, aligning Dart with the future of the Web.
 
+The following table maps Dart's new JS and web interop solutions to
+their past counterparts:
+
+<div class="table-wrapper" markdown="1">
+
+| New static interop libraries        | Previous libraries                       |
```
Comment (kevmoo) [inline src/content/interop/js-interop/index.md:25]: We're trying NOT to use the phrase `static interop`, right? @srujzs

**Comment by kevmoo** [comment]
CC @mit-mit – who wanted more goodness here!

File: src/content/interop/js-interop/index.md (hunk 1 above)
Comment (MaryaBelanger) [reply src/content/interop/js-interop/index.md:25]: The [package:web](https://pub.dev/packages/web) pub page says "Lightweight browser API bindings built around JS static interop" in the Metadata section of the sidebar. Not saying that's the end-all, but maybe that will need to change too?

File: src/content/interop/js-interop/index.md [hunk 2]
```diff
@@ -10,12 +10,44 @@ JavaScript apps and libraries, as well as browser APIs, using `dart:js_interop`.
...
+|-------------------------------------|------------------------------------------|
+| [`package:web`][] | [`dart:html`][] <br> [`dart:svg`][] <br> [`dart:indexed_db`][] <br> [`dart:web_audio`][] <br> [`dart:web_gl`][] |
```
[the remark this answers is not in the document]
Comment (MaryaBelanger) [reply src/content/interop/js-interop/index.md:27]: Sort them how? I checked the [API home page](https://api.dart.dev/dev/3.4.0-204.0.dev/index.html) and it looks like `dart:svg` should come after `indexed`, but that's the only change I see. Lmk if you're thinking something else (alphabetically?)

File: src/content/interop/js-interop/index.md (hunk 1 above)
Comment (kevmoo) [reply src/content/interop/js-interop/index.md:25]: Yeah, just alphabetical

File: src/content/interop/js-interop/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevmoo) [reply src/content/interop/js-interop/index.md:27]: It SHOULD, right @srujzs ?

**Review by srujzs — APPROVED** [review]
Thanks Marya!

File: src/content/interop/js-interop/index.md [hunk 3]
```diff
@@ -10,12 +10,44 @@ JavaScript apps and libraries, as well as browser APIs, using `dart:js_interop`.
...
+but also enables [Wasm][] support, aligning Dart with the future of the Web.
 
+The following table maps Dart's new JS and web interop solutions to
+their past counterparts:
+
+<div class="table-wrapper" markdown="1">
+
+| New static interop libraries        | Previous libraries                       |
+|-------------------------------------|------------------------------------------|
+| [`package:web`][] | [`dart:html`][] <br> [`dart:indexed_db`][] <br> [`dart:svg`][] <br> [`dart:web_audio`][] <br> [`dart:web_gl`][] |
```
[the remark this answers is not in the document]
Comment (MaryaBelanger) [reply src/content/interop/js-interop/index.md:27]: I couldn't find a link to any api docs, so I'm going to skip it, but if you have one lmk!

**Pull request merged**